packages/components/src/unit-control/utils.js

```diff
@@ -9,6 +9,17 @@ import { isEmpty } from 'lodash';
 import { __, _x } from '@wordpress/i18n';
 import { Platform } from '@wordpress/element';
 
+/**
+ * An object containing the details of a unit.
+ *
+ * @typedef {Object} WPUnitControlUnit
+ * @property {string}        value       The value for the unit, used in a CSS value (e.g `px`).
+ * @property {string}        label       The label used in a dropdown selector for the unit.
+ * @property {string|number} [default]   Default value for the unit, used when switching units.
+ * @property {string}        [a11yLabel] An accessible label used by screen readers.
+ * @property {number}        [step]      A step value used when incrementing/decrementing the value.
+ */
+
 const isWeb = Platform.OS === 'web';
 
 const allUnits = {
```

Comment:
Adding this line should also type `allUnits`, `ALL_CSS_UNITS`, `CSS_UNITS` and `DEFAULT_UNIT` 

```suggestion
/** @type {Record<string, WPUnitControlUnit>} */
const allUnits = {
```

Reply:
Nice one! Committed.